packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketBuyForm.tsx

```diff
@@ -0,0 +1,521 @@
+import { AmountLimits } from 'src/types/wallet/coinmarketCommonTypes';
+import { useCoinmarketBuyFormDefaultValues } from './useCoinmarketBuyFormDefaultValues';
+import { CoinmarketTradeBuyType, UseCoinmarketFormProps } from 'src/types/coinmarket/coinmarket';
+import {
+    addIdsToQuotes,
+    filterQuotesAccordingTags,
+} from 'src/utils/wallet/coinmarket/coinmarketUtils';
+import {
+    CoinmarketBuyFormContextProps,
+    CoinmarketBuyFormProps,
+} from 'src/types/coinmarket/coinmarketForm';
+import {
+    getFilteredSuccessQuotes,
+    useCoinmarketCommonOffers,
+} from '../offers/useCoinmarketCommonOffers';
+import * as coinmarketInfoActions from 'src/actions/wallet/coinmarketInfoActions';
+import * as coinmarketCommonActions from 'src/actions/wallet/coinmarket/coinmarketCommonActions';
+import * as coinmarketBuyActions from 'src/actions/wallet/coinmarketBuyActions';
+import * as routerActions from 'src/actions/suite/routerActions';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import { isDesktop } from '@trezor/env-utils';
+import { SET_MODAL_CRYPTO_CURRENCY } from 'src/actions/wallet/constants/coinmarketCommonConstants';
+import useCoinmarketPaymentMethod from 'src/hooks/wallet/coinmarket/form/useCoinmarketPaymentMethod';
+import { useBitcoinAmountUnit } from 'src/hooks/wallet/useBitcoinAmountUnit';
+import { useCoinmarketNavigation } from 'src/hooks/wallet/useCoinmarketNavigation';
+import { FORM_PAYMENT_METHOD_SELECT } from 'src/constants/wallet/coinmarket/form';
+import { useCoinmarketSatsSwitcher } from 'src/hooks/wallet/coinmarket/form/useCoinmarketSatsSwitcher';
+import { Network } from '@suite-common/wallet-config';
+import { cryptoToNetworkSymbol } from 'src/utils/wallet/coinmarket/cryptoSymbolUtils';
+
+const useCoinmarketBuyForm = ({
+    selectedAccount,
+    pageType = 'form',
+}: UseCoinmarketFormProps): CoinmarketBuyFormContextProps => {
+    const type = 'buy';
+    const isPageOffers = pageType === 'offers';
+    const dispatch = useDispatch();
+    const { addressVerified, buyInfo, isFromRedirect, quotes, quotesRequest } = useSelector(
+        state => state.wallet.coinmarket.buy,
+    );
+    const {
+        callInProgress,
+        account,
+        selectedQuote,
+        timer,
+        device,
+        setCallInProgress,
+        setSelectedQuote,
+        checkQuotesTimer,
+    } = useCoinmarketCommonOffers<CoinmarketTradeBuyType>({ selectedAccount, type });
+    const { paymentMethods, getPaymentMethods, getQuotesByPaymentMethod } =
+        useCoinmarketPaymentMethod<CoinmarketTradeBuyType>();
+    const {
+        saveTrade,
+        saveQuotes,
+        setIsFromRedirect,
+        openCoinmarketBuyConfirmModal,
+        addNotification,
+        saveTransactionDetailId,
+        verifyAddress,
+        submitRequestForm,
+        goto,
+        savePaymentMethods,
+        saveQuoteRequest,
+        saveCachedAccountInfo,
```

Comment:
Consider to remove saveCachedAccountInfo and related code in reducer and the action (also test). Check if we are using it or not.